pages/product/[code].js

```diff
@@ -183,9 +234,34 @@ export default function DetailPage() {
         <>
           <StyledAllCards>
             <StyledProductCard>
-              <StyledProductName>
-                {data.product.product_name}{" "}
-              </StyledProductName>
+              <StyledProductName>{data.product.product_name}</StyledProductName>
+              <StyledButtonDiv>
+                <StyledSaveButton
+                  onClick={() => {
+                    if (
+                      !savedProducts.some((product) => product.code === code)
+                    ) {
+                      setSavedProducts([
+                        {
+                          name: data.product.product_name,
+                          url: data.product.image_front_url,
+                          code: code,
+                        },
+                        ...savedProducts,
+                      ]);
+                    }
+                  }}
+                >
+                  <SVGIcon
+                    variant={
+                      savedProducts.some((product) => product.code === code)
```

Comment:
you check for the existence of the product twice. Please refactor it into a constant above the return value and use that variable here and in line 242 instead.

Reply:
done .thanks